### PR TITLE
Don't suggest unstable items on stable toolchain

### LINKED ISSUE #3020: Do not propose unstable features on autocomplete for stable toolchain
In the project with stable toolchain used, `Some(22).ex<|>` proposes `expect_none` [method](https://doc.rust-lang.org/std/option/enum.Option.html#method.expect_none) which is unstable currently.

image

I would expect RA not to propose unstable APIs when a stable toolchain is used.

### PR DESCRIPTION
**Pull request by lowr**
Closes #3020

This PR implements stability check in `ide-completion` so that unstable items are only suggested if you're on nightly toolchain.

It's a bit unfortunate `CompletionContext::check_stability()` is spammed all over the crate, but we should call it before building `CompletionItem` as you cannot get attributes on the item it's completing from that struct. I looked up every callsite of `Builder::add_to()`, `Completions::add[_opt]()`, and`Completions::add_all()` and inserted the check wherever necessary.

The tests are admittedly incomplete in that I didn't add tests for every kind of item as I thought that would be too big and not worthwhile. I copy-pasted some existing basic tests in every test module and adjusted them.

### CODE REVIEW

**Review by Veykril — APPROVED** [review]

File: crates/ide-completion/src/completions/flyimport.rs [hunk 1]
```diff
@@ -315,8 +317,10 @@ fn import_on_the_fly_pat_(
             .into_iter()
             .filter(ns_filter)
             .filter(|import| {
-                !ctx.is_item_hidden(&import.item_to_import)
-                    && !ctx.is_item_hidden(&import.original_item)
+                let item = &import.item_to_import;
+                !ctx.is_item_hidden(item)
```
Comment (hecatia-elegua) [inline crates/ide-completion/src/completions/flyimport.rs:321]: Did you want to do `!ctx.is_item_hidden(&import.original_item)` here?

File: crates/ide-completion/src/completions/flyimport.rs [hunk 2]
```diff
@@ -267,8 +267,10 @@ fn import_on_the_fly(
...
+                    && !ctx.is_item_hidden(item)
```
Comment (hecatia-elegua) [inline crates/ide-completion/src/completions/flyimport.rs:272]: here as well

File: crates/ide-completion/src/completions/flyimport.rs (hunk 1 above)
Comment (lowr) [reply crates/ide-completion/src/completions/flyimport.rs:321]: Oops, you're right! Thanks for catching this. Will file another PR shortly!

**Comment by lnicola** [comment]
Um, how do I make this work?

image

**Comment by lowr** [comment]
@lnicola Looks like our release channel detection procedure is wrong. I'm sorry I didn't notice this before landing, when I tested in my local env it worked, but I guess that was before I added `None` [here].

Fix on the way!

[here]: https://github.com/rust-lang/rust-analyzer/pull/14549/files#diff-563c0914dd89f1c5e6ca3ae5eb22105f0bcd42c15ae26580a4e7a25dc10d9a96R635-R639

**Comment by lnicola** [comment]
@lowr no idea why, I have a vanilla rustup setup.

**Pull request merged**